### PR TITLE
tools(vagrant): add box for building Windows containers

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Found a turnkey image with Docker for Windows configured, where it's easy to build Windows containers.

### CODE REVIEW

**Review by sumo-drosiek — COMMENTED** [review]
I will try it on Monday

File: vagrant/windows/README.md [hunk 1]
```diff
@@ -47,3 +47,14 @@ make build
...
+
+## Building Windows containers
+
+Use the [docker](docker/) machine, as it comes with Docker for Windows installed and configured. After building
+the application, you can build a Windows container the following way:
+
+```bash
+cp otelcolbuilder/cmd/otelcol-sumo.exe .
+make build-push-container-windows-dev PLATFORM=windows/amd64/ltsc2022
+docker run public.ecr.aws/sumologic/sumologic-otel-collector-dev:latest-windows-amd64-ltsc2022 --version
```
Comment (sumo-drosiek) [inline vagrant/windows/README.md:59]:
```
+ docker build --file Dockerfile_windows --build-arg BUILD_TAG=latest --build-arg BASE_IMAGE_TAG=ltsc2022 --build-arg BUILDKIT_INLINE_CACHE=1 --platform=windows/amd64 --tag public.ecr.aws/sumologic/sumologic-otel-collector-dev:latest-windows-amd64-ltsc2022 .
unable to prepare context: unable to evaluate symlinks in Dockerfile path: CreateFile \\vboxsvr\sumologicDockerfile_windows: The network name cannot be found.
make: *** [Makefile:302: _build-container-multiplatform] Error 1
```

Comment (swiatekm-sumo) [reply vagrant/windows/README.md:59]: Have you modified the Vagrantfile? That looks like a shared folder problem of some kind. Does it work if you copy the repo to the VM disk?

Comment (sumo-drosiek) [reply vagrant/windows/README.md:59]: It works when I change workdir to `\\vboxsrv\sumologic\`, and also works if using `bash`

Comment (swiatekm-sumo) [reply vagrant/windows/README.md:59]: Allright, can we merge it as-is then, and work through this problem afterwards?

Comment (sumo-drosiek) [reply vagrant/windows/README.md:59]: Fine for me as I know the solution :)

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**